### ISSUE TITLE
Some questions about the code.

### ISSUE THREAD

**Issue by hljjjmssyh** [dm-thu/GraphSGAN#3]
Hi!
Thanks for your contribution. I want to ask you some questions about the code.
1. You mentioned the classifier output (M+1) units, but you defined M units in your code.
2. I noticed that you added a weight to the loss_unsuperviesd. Is this a hyperparameter?
![image](https://user-images.githubusercontent.com/24326757/65813553-86766580-e209-11e9-961d-f5d63f1789e5.png)
Thank you!

**Comment by Sleepychord** [comment]
Hi, this is a technique from improved gan. As showed in this latex picture, 

<img width="956" alt="image" src="https://user-images.githubusercontent.com/9153807/65815396-6783cd80-e221-11e9-9f7a-13fb34de1521.png">

**Comment by hljjjmssyh** [comment]
Ok. Thanks for your explanation.
Why real_data loss function multiply by 0.5?

**Comment by Sleepychord** [comment]
To balance the size of supervised and unsupervised data. We have a batch of unlabeled real data and a batch of fake data for unsupervised learning, but only a batch of labeled data for supervised learning.
